javav2/usecases/creating_photo_analyzer_async/Readme.md

```diff
@@ -0,0 +1,1532 @@
+
+```java
+     package com.example.photo;
+
+      import org.springframework.stereotype.Component;
+      import org.w3c.dom.Document;
+      import org.w3c.dom.Element;
+      import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+      import software.amazon.awssdk.core.ResponseBytes;
+      import software.amazon.awssdk.core.async.AsyncRequestBody;
+      import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+      import software.amazon.awssdk.regions.Region;
+      import software.amazon.awssdk.services.s3.S3AsyncClient;
+      import software.amazon.awssdk.services.s3.model.*;
+      import javax.xml.parsers.DocumentBuilder;
+      import javax.xml.parsers.DocumentBuilderFactory;
+      import javax.xml.parsers.ParserConfigurationException;
+      import javax.xml.transform.Transformer;
+      import javax.xml.transform.TransformerException;
+      import javax.xml.transform.TransformerFactory;
+      import javax.xml.transform.dom.DOMSource;
+      import javax.xml.transform.stream.StreamResult;
+      import java.io.StringWriter;
+      import java.time.Instant;
+      import java.util.ArrayList;
+      import java.util.List;
+      import java.util.ListIterator;
+      import java.util.concurrent.CompletableFuture;
+      import java.util.concurrent.atomic.AtomicReference;
+
+      @Component
+      public class S3Service {
+
+       S3AsyncClient s3AsyncClient;
+
+      // Create a  thread-safe, object.
+       private S3AsyncClient getClient() {
+
+        Region region = Region.US_WEST_2;
+        return S3AsyncClient.builder()
+                .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+                .region(region)
+                .build();
+       }
+
+        //Return a byte[] that represents the object.
+        public byte[] getObjectBytes (String bucketName, String keyName) {
+         s3AsyncClient = getClient();
+         final AtomicReference<byte[]> reference = new AtomicReference<>();
+
+          try {
+
+            GetObjectRequest objectRequest = GetObjectRequest.builder()
+                    .bucket(bucketName)
+                    .key(keyName)
+                    .build();
+
+            // Get the Object from the Amazon S3 bucket using the Amazon S3 Async Client
+            final CompletableFuture<ResponseBytes<GetObjectResponse>>[] futureGet = new CompletableFuture[]{s3AsyncClient.getObject(objectRequest,
+                    AsyncResponseTransformer.toBytes())};
+
+            futureGet[0].whenComplete((resp, err) -> {
+                try {
+
+                    if (resp != null) {
+
+                        //  Set the AtomicReference object
+                         reference.set(resp.asByteArray());
+
+                    } else {
+                        err.printStackTrace();
+                    }
+                } finally {
+
+                    // Only close the client when you are completely done with it
```

Comment:
Once again, you need a period at the end of the sentence.

Reply:
fixed